postgresql-cdc/src/main/scala/akka/stream/alpakka/postgresqlcdc/PgSQLChangeDataCaptureSourceStage.scala

```diff
@@ -0,0 +1,214 @@
+/** Settings for PostgreSQL CDC
+ *
+ * @param connectionString PostgreSQL JDBC connection string
+ * @param slotName Name of the "logical decoding" slot
+ * @param maxItems Specifies how many rows are fetched in one batch
+ * @param duration Duration between polls
+ */
+final case class PostgreSQLChangeDataCaptureSettings(connectionString: String,
+                                                     slotName: String,
+                                                     maxItems: Int = 128,
+                                                     duration: FiniteDuration = 2000 milliseconds)
+
+sealed trait Change {
+  val schemaName: String
+  val tableName: String
+}
+
+case class Field(columnName: String, columnType: String, value: String)
+
+case class RowInserted(schemaName: String, tableName: String, fields: Set[Field]) extends Change
+
+case class RowUpdated(schemaName: String, tableName: String, fields: Set[Field]) extends Change
+
+case class RowDeleted(schemaName: String, tableName: String, fields: Set[Field]) extends Change
+
+case class ChangeSet(transactionId: Long, changes: Set[Change]) // TODO: add timestamp
+
+private[postgresqlcdc] object PgSQLChangeDataCaptureSourceStage {
+
+  import Grammar._
+
+  /** Represents a row in the table we get from PostgreSQL when we query
+   * SELECT * FROM pg_logical_slot_get_changes(..)
+   */
+  case class SlotChange(transactionId: Long, location: String, data: String)
+
+  def parseProperties(properties: String): Set[Field] =
+    Property
+      .findAllMatchIn(properties)
+      .collect {
+        case regexMatch if regexMatch.groupCount == 3 =>
+          // note that there is group 0 that denotes the entire match - and it is not included in the groupCount
+          val columnName: String = regexMatch.group(1)
+          val columnType: String = regexMatch.group(2)
+          val value: String = regexMatch.group(3) match {
+            case SingleQuotedString2(content) => content
+            case other => other
+          }
+          Field(columnName, columnType, value)
+      }
+      .toSet
+
+  def transformSlotChanges(slotChanges: Set[SlotChange]): Set[ChangeSet] = {
+
+    slotChanges.groupBy(_.transactionId).map {
+
+      case (transactionId: Long, slotChanges: Set[SlotChange]) =>
+        val changes: Set[Change] = slotChanges.collect {
+
+          case SlotChange(_, _, ChangeStatement(schemaName, tableName, "UPDATE", properties)) =>
+            RowUpdated(schemaName, tableName, parseProperties(properties))
+
+          case SlotChange(_, _, ChangeStatement(schemaName, tableName, "DELETE", properties)) =>
+            RowDeleted(schemaName, tableName, parseProperties(properties))
+
+          case SlotChange(_, _, ChangeStatement(schemaName, tableName, "INSERT", properties)) =>
+            RowInserted(schemaName, tableName, parseProperties(properties))
+        }
+
+        ChangeSet(transactionId, changes)
+
+    }
+  }.filter(_.changes.nonEmpty).toSet
+
+  object Grammar {
```

Comment:
some of the regular expressions below need to be hardened, to catch some edge cases ;-)